### ISSUE TITLE
curl errors

### ISSUE THREAD

**Issue by sebastian-fox** [ropensci/fingertipsR#44]
Some local authorities are getting curl errors. Likely to be a result of proxy issues. I have been pointed towards this issue: https://github.com/rOpenGov/eurostat/issues/99

Also, workaround created by one LA doing the following:
```
Chosen_Profile = 87 # Profile for LAPE
Chosen_Ind = 92778 #  This is the indicator for percentage of adults who drink over 14 units per week
Chosen_area = 102 # This is the code for upper tier LA, 101 is for lower tier LA, 7 for GPs, 6 for region
Chosen_parent_area = 6

LAPE_over14  <- read_csv(url(paste("https://fingertips.phe.org.uk/api/all_data/csv/by_indicator_id?indicator_ids=", Chosen_Ind,"&child_area_type_id=", Chosen_area ,"&parent_area_type_id=",Chosen_parent_area,"&profile_id=", Chosen_Profile, sep = "")))
```

**Comment by p-turner** [comment]
I struggled to traverse a proxy server at work in order to use fingertipsR.  However, I found that the following works for me but if your proxy server settings are different this may not work for you:

Either explicitly use the proxy server url and port as below:

**set_config(use_proxy(url = "http://**<**proxy_url**>**", port = **<**proxy_port**>**, username = "", password = "", auth = "ntlm" ), override = TRUE)**

_(Requires httr package)_

or use a generic method that retrieves the url and port information from the proxy server:

**proxy <- ie_get_proxy_for_url(target_url = "https://fingertips.phe.org.uk")
proxy_url <- substr(proxy, 1, nchar(proxy) - 5)
proxy_port <- as.numeric(substr(proxy, nchar(proxy) - 3, nchar(proxy)))
proxy_config <- use_proxy(url = proxy_url, port = proxy_port, username = "", password = "", auth = "ntlm" )
set_config(proxy_config, override = TRUE)**

_(Requires curl and httr packages)_

Once you have retrieved all the data you require from the Fingertips API using the fingertipsR package then use:

**reset_config()** # to reset to default configuration.

Please note:
1.  I contacted an IT colleague at work before doing any of the above.  I would strongly advise anyone to do the same in order not to risk transgressing your organisation's IT Acceptable Use Policy.
2.  I did not have to use my work user name or password to make this work but I did have to use an empty string for each rather than depend on the default NULL values.  Do not use your work login credentials without seeking the appropriate IT advice.
3. "ntlm" is a relatively old Microsoft authentication method.  It may be that this is the default for most proxy servers running Windows but I cannot be sure.

**Comment by sebastian-fox** [comment]
Thanks for looking into this @p-turner . I will speak to people to understand whether I can build this into the package

**Comment by sebastian-fox** [comment]
Hi @p-turner - I think I have managed to solve this now (in 26c5b6b79f5c5e594175109471e11287ab91ea2d)

It has been tested by one LA, so hopefully this works for all but am keen to hear feedback if not.

For now, you will need to install the package as follows, but the changes will be available when I next submit to CRAN (version 0.1.7) which should be soon:

* Go to this link: https://github.com/ropensci/fingertipsR 
* Click “Clone or download”
* Download zip
* Extract the folder from the zipped file
* Then from within R:

```
source <- devtools:::source_pkg("C:/path/to/fingertipsR-master")
install.packages(source, repos = NULL, type = "source")
```

Many thanks to Una Geary for helping test this and to Alex Betts for helping solve the issue.

**Comment by sebastian-fox** [comment]
And thanks to @p-turner too!
